src/middleware.ts

```diff
@@ -2,14 +2,28 @@ import { type NextRequest, NextResponse } from "next/server";
 
 import { Routes } from "./lib/routes";
 import { getSession } from "./utils/supabase/getSession";
+import { getLoggedInUser } from "./utils/getLoggedInUser";
+import { MemberRole } from "./constant/constant";
+
+const restrictedPaths: Routes[] = [Routes.AttendanceStatus];
 
 export default async function updateSession(request: NextRequest) {
   const isTokenValid = await getSession();
-  const requestedPath = request.nextUrl.pathname;
+  const requestedPath = request.nextUrl.pathname as Routes;
+  const loginUser = await getLoggedInUser();
+
+  const isRestrictedPath = restrictedPaths.includes(requestedPath);
 
   if (!isTokenValid && requestedPath !== Routes.Login) {
     return NextResponse.redirect(new URL(Routes.Login, request.url));
   }
+  if (
```

Comment:
nice

Reply:
Thank you for appreciating and for helping me out here